File: ДЗ 2/ch-4-1-step-10.swift
import Foundation

func read2() -> (Int, Int) {
  let line = readLine()!.split(separator: " ")
  let int0 = Int(line[0])!
  let int1 = Int(line[1])!
  return (int0, int1)
}

struct Item {
  let price: Int
  let size: Int
  var partCost: Double {
    return Double(price) / Double(size)
  }
}

let (count, capacity) = read2()
var items: [Item] = []
for _ in 0..<count {
  let (price, size) = read2()
  let item = Item(price: price, size: size)
  items.append(item)
}

items.sort(by: { $0.partCost > $1.partCost })

var remainingCapacity = capacity
var totalPrice: Double = 0
for item in items {
  let partsCount = min(item.size, remainingCapacity)
  remainingCapacity -= partsCount
  totalPrice += Double(partsCount) * item.partCost
}

print(String(format: "%.3f", totalPrice))

File: ДЗ 2/ch-4-1-step-10.py
def read2():
    line = input().split()
    int0 = int(line[0])
    int1 = int(line[1])
    return (int0, int1)

class Item:
    def __init__(self, price, size):
        self.price = price
        self.size = size

    @property
    def partCost(self):
        return float(self.price) / float(self.size)

count, capacity = read2()
items = []
for _ in range(count):
    price, size = read2()
    item = Item(price, size)
    items.append(item)

items.sort(key=lambda x: x.partCost, reverse=True)

remainingCapacity = capacity
totalPrice = 0
for item in items:
    partsCount = min(item.size, remainingCapacity)
    remainingCapacity -= partsCount
    totalPrice += float(partsCount) * item.partCost

print("{:.3f}".format(totalPrice))
